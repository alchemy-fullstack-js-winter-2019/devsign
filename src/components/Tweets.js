import React from 'react';
import PropTypes from 'prop-types';
import Tweet from './Tweet';

function Tweets({ tweets }) {
  const tweetList = tweets.map((tweet, i) => {
    return <li key={i}><Tweet
      profileImage={tweet.profileImage}
      name={tweet.name} 
      text={tweet.text}
      handle={tweet.handle}
    /></li>;
  });
  return (
    <section>
      <ul>
        {tweetList}
      </ul>
    </section>
  );
}

Tweets.propTypes = {
  tweets: PropTypes.array.isRequired
};

export default Tweets;
