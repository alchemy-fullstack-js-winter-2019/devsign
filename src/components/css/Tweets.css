.tweetItems {
  list-style-type: none;
}