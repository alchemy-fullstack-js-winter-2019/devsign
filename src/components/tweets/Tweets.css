.Tweets {
  list-style: none;
  
  
  li {
    border: 2px solid black;
    margin: 2vh 2vw;
    width: 70vw;
  }
}