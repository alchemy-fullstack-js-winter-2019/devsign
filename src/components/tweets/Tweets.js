import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Tweet from './Tweet';

function Tweets({ tweets }) {
  const tweetsList = tweets.map(tweet =>
    <li key={tweet.id}>
      <Link to={`/user/${tweet.userId}`}>
        <Tweet tweet={tweet} />
      </Link>
    </li>
  );

  return (
    <>
      <h2>Latest Tweets:</h2>
      <ul>{tweetsList}</ul>
    </>
  );
}

Tweets.propTypes = {
  tweets: PropTypes.array.isRequired
};

export default Tweets;
