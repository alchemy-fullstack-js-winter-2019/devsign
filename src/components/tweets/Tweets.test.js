import React from 'react';
import renderer from 'react-test-renderer';
import Tweets from './Tweets';
import { MemoryRouter } from 'react-router-dom';

describe('Tweets', () => {
  it('displays a tweet', () => {
    const tweets = [
      {
        id: 1,
        userId: 1,
        img: 'https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060',
        text: 'twitter you twit'
      },
      {
        id: 2,
        userId: 2,
        img: 'https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060',
        text: 'twitter you twit twit'
      },
      {
        id: 3,
        userId: 3,
        img: 'https://cdn.shopify.com/s/files/1/1061/1924/products/Thinking_Face_Emoji_large.png?v=1480481060',
        text: 'twitter you twit twit twitter twit'
      }
    ];

    const tree = renderer.create(
      <MemoryRouter>
        <Tweets tweets={tweets} />
      </MemoryRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
