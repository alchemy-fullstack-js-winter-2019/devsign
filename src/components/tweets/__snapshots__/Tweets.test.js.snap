// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tweets matches a snapshot 1`] = `
<section
  id="all-tweets"
>
  <article
    style={
      Object {
        "padding": ".8em",
      }
    }
  >
    <img
      alt="profile-pic"
      className="img"
      src="../../assets/circle_sm.png"
    />
    <h3
      className="h3"
    >
      Aasdfa
    </h3>
    <span
      className="span"
    >
      Lorem ipsum dolor sit amet, uioua adipiscing elit, sed do eiusmod. #asdfasdf #tehklj
    </span>
  </article>
  <article
    style={
      Object {
        "padding": ".8em",
      }
    }
  >
    <img
      alt="profile-pic"
      className="img"
      src="../../assets/circle_sm.png"
    />
    <h3
      className="h3"
    >
      Tio
    </h3>
    <span
      className="span"
    >
      Ut enim ad minim veniam, @jlakdsf quis nostrud #lkjasfd
    </span>
  </article>
</section>
`;
